rajawali/src/main/java/org/rajawali3d/materials/Material.java

```diff
@@ -1073,7 +1073,12 @@ public void setModelMatrix(Matrix4 modelMatrix) {
         mModelMatrix = modelMatrix;//.getFloatValues();
         mVertexShader.setModelMatrix(mModelMatrix);
 
-        mNormalMatrix.setAll(modelMatrix).setToNormalMatrix();
+        mNormalMatrix.setAll(modelMatrix);
+        try {
+            mNormalMatrix.setToNormalMatrix();
+        } catch (IllegalStateException exception) {
+            // modelMatrix is degenerate (zero scale)
```

Comment:
Or, rather than computing the determinant and getting the exception, perhaps we could just check for zero scale before even calling setToNormal()?